homeassistant/components/touchline/climate.py

```diff
@@ -95,8 +124,91 @@ def target_temperature(self):
         """Return the temperature we try to reach."""
         return self._target_temperature
 
+    @property
+    def current_operation_mode(self):
+        """Return the current operation mode."""
+        return self._current_operation_mode
+
+    @property
+    def current_week_program(self):
+        """Return the current week program."""
+        return self._current_week_program
+
+    @property
+    def current_preset_mode(self):
+        """Return the current preset mode."""
+        return self._current_preset_mode
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self):
+        """Return available preset modes."""
+        return PRESET_MODES
+
+    def set_preset_mode(self, preset_mode):
+        """Set new target preset mode."""
+        mode = self.map_mode_hass_touchline(preset_mode)
+        program = self.map_program_hass_touchline(preset_mode)
+        self.unit.set_operation_mode(mode)
+        self.unit.set_week_program(program)
+        self._current_operation_mode = mode
+        self._current_week_program = program
+        self._current_preset_mode = preset_mode
```

Comment:
We don't need to update any state holding instance attributes since this is a polling entity and `update` will be called at the end of the service call.